bears/general/FilenameBear.py

```diff
@@ -66,6 +74,24 @@ def run(self, filename, file,
         try:
             new_name = self._naming_convention[file_naming_convention](
                 filename_without_extension)
+            messages = []
+
+            if new_name != filename_without_extension:
+                messages.append(
+                    'Filename does not follow {} naming-convention.'.format(
+                        file_naming_convention))
+
+            if not filename_without_extension.startswith(filename_prefix):
+                new_name = filename_prefix + new_name
+                messages.append(
+                    'Filename does not use the {!r} prefix.'.format(
+                        filename_prefix))
+
+            if not filename_without_extension.endswith(filename_suffix):
+                new_name = new_name + filename_suffix
+                messages.append(
+                    'Filename does not use the {!r} suffix.'.format(
+                        filename_suffix))
```

Comment:
`try`-blocks should be left as early as possible, as we want to fail fast on other errors and don't want them to slip through :)